### ISSUE TITLE
Extra content is highlighted when using '*' search

### ISSUE THREAD

**Issue by jacobdufault** [VSCodeVim/Vim#1844]
* Click *thumbs-up* 👍 on this issue if you want it!
* Click *confused* 😕 on this issue if not having it makes VSCodeVim unusable. 

The VSCodeVim team prioritizes issues based on reaction count. 

--------

**BUG REPORT**

**What happened**:

With this text,

```
foo  f
foo  f
```

Put the cursor on `foo` and hit `*`. Notice that the whitespace after `foo` is highlighted and the inactive `foo` has a double-highlight, one of which extends past the `foo`.

![image](https://user-images.githubusercontent.com/356714/27200854-495e40bc-51d0-11e7-8807-29ed21c6e383.png)

**Comment by Chillee** [comment]
I assume this is with the `vim.hlsearch` set to true right?

**Comment by xconverge** [comment]
Yea I noticed this today as well

**Comment by jacobdufault** [comment]
Yep, `vim.hlsearch` needs to be true. Note that this has been a long-standing issue (ie, months).

**Comment by xconverge** [comment]
Not that it works fine with / so this is isolated to *
